src/main/java/labs/introtoprogramming/lab5/graphics/ArrayRaster.java

```diff
@@ -0,0 +1,61 @@
+package labs.introtoprogramming.lab5.graphics;
+
+import java.awt.Color;
+import java.awt.Image;
+import java.awt.image.BufferedImage;
+
+public class ArrayRaster implements Raster {
+
+  private final int width;
+  private final int height;
+
+  private byte[][] red;
+  private byte[][] green;
+  private byte[][] blue;
+
+  public ArrayRaster(int width, int height) {
+    this.width = width;
+    this.height = height;
+    this.red = new byte[width][height];
+    this.green = new byte[width][height];
+    this.blue = new byte[width][height];
+  }
+
+  public Image toImage() {
+    BufferedImage image = new BufferedImage(getWidth(), getHeight(), BufferedImage.TYPE_INT_RGB);
+    for (int x = 0; x < width; x++) {
+      for (int y = 0; y < height; y++) {
+        image.setRGB(x, y, getPixelRGB(x, y));
+      }
+    }
+    return image;
+  }
+
+  public void setPixel(int x, int y, int rgb) {
+    this.red[x][y] = (byte) (rgb >> 16);
+    this.green[x][y] = (byte) ((rgb >> 8) & 0xFF);
+    this.blue[x][y] = (byte) (rgb & 0xFF);
+  }
+
+  public void setPixel(int x, int y, byte red, byte green, byte blue) {
+    this.red[x][y] = red;
+    this.green[x][y] = green;
+    this.blue[x][y] = blue;
+  }
+
+  public Color getPixelColor(int x, int y) {
+    return new Color(red[x][y] & 255, green[x][y] & 255, blue[x][y] & 255);
+  }
+
+  public int getPixelRGB(int x, int y) {
+    return 0xFF << 24 | (red[x][y] << 16) | (green[x][y] << 8) | blue[x][y];
```

Comment:
I found here [https://stackoverflow.com/questions/13154715/convert-3-bytes-to-int-in-java](url) that you need firstly to use AND operator before shifting because bytes are signed in Java. Am I wrong?